formatter.js

```diff
@@ -0,0 +1,54 @@
+/*
+  Custom eslint formatter for eslint-plugin-i18n-json to allow better error message display.
+  Heavily inspired from https://github.com/sindresorhus/eslint-formatter-pretty.
+*/
+
+const chalk = require('chalk');
+const plur = require('plur');
+const logSymbols = require('log-symbols');
+const indentString = require('indent-string');
+const path = require('path');
+
+const CWD = process.cwd();
+
+const formatter = (results) => {
+  const total = {
+    errors: 0,
+    warnings: 0,
+  };
+  const outputs = results.map(({
+    filePath,
+    messages,
+    errorCount,
+    warningCount,
+  }) => {
+    if (errorCount + warningCount === 0) {
+      return '';
+    }
+    total.errors += errorCount;
+    total.warnings += warningCount;
+    const relativePath = path.relative(CWD, filePath);
+    const header = `${chalk.underline.white(relativePath)}`;
+
+    messages.sort((a, b) => b.severity - a.severity);
+
+    const messagesOutput = messages.map(({ ruleId, severity, message }) => {
+      let messageHeader = `${severity === 1 ? `${logSymbols.warning} ${chalk.inverse.yellow(' WARNING ')}` : `${logSymbols.error} ${chalk.inverse.red(' ERROR ')}`}`;
+      messageHeader += ` ${chalk.white(`(${ruleId})`)}`;
```

Comment:
does this evaluation to the same thing?  (the nested template expressions are hard to parse)

```js
messageHeader += ' ' + chalk.white(`(${ruleId})`);
```

Reply:
agreed, will update. originally had it that way, but was adhering to https://eslint.org/docs/rules/no-useless-concat